lagom14-scala-conductr-bundle-lib/src/test/scala/com/typesafe/conductr/bundlelib/lagom/scaladsl/ConductRServiceLocatorSpecWithEnv.scala

```diff
@@ -0,0 +1,122 @@
+package com.typesafe.conductr.bundlelib.lagom.scaladsl
+
+import java.net.{ InetSocketAddress, URI => JavaURI }
+
+import akka.actor._
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.headers.{ CacheDirectives, Location, `Cache-Control` }
+import akka.http.scaladsl.model.{ HttpEntity, HttpResponse, StatusCodes }
+import akka.http.scaladsl.server.Directives._
+import akka.stream.ActorMaterializer
+import akka.testkit.TestProbe
+import com.lightbend.lagom.internal.client.{ CircuitBreakerConfig, CircuitBreakerMetricsProviderImpl, CircuitBreakers }
+import com.lightbend.lagom.internal.spi.CircuitBreakerMetricsProvider
+import com.lightbend.lagom.scaladsl.api.{ Descriptor, ServiceAcl, ServiceInfo }
+import com.lightbend.lagom.scaladsl.client.CircuitBreakersPanel
+import com.typesafe.conductr.bundlelib.play.api.{ Env => PlayEnv }
+import com.typesafe.conductr.bundlelib.scala.{ URI, URL }
+import com.typesafe.conductr.lib.AkkaUnitTestWithFixture
+import play.api.libs.ws.ahc.AhcWSComponents
+import play.api.routing.Router
+import play.api.test.Helpers._
+import play.api.{ ApplicationLoader, BuiltInComponentsFromContext, Environment }
+
+import scala.collection.immutable
+import scala.concurrent.{ Await, Future }
+import scala.util.{ Failure, Success }
+
+class ConductRServiceLocatorSpecWithEnv extends AkkaUnitTestWithFixture("ConductRServiceLocatorSpecWithEnv") {
+
+  def systemFixture(f: this.FixtureParam) = new {
+    implicit val system = f.system
+    implicit val timeout = f.timeout
+    implicit val mat = ActorMaterializer.create(system)
+    val app = new BuiltInComponentsFromContext(ApplicationLoader.createContext(Environment.simple())) with AhcWSComponents with ConductRServiceLocatorComponents {
+      override lazy val actorSystem = system
+      override lazy val materializer = mat
+      override lazy val executionContext = actorSystem.dispatcher
+      override lazy val router = Router.empty
+      override lazy val circuitBreakerMetricsProvider: CircuitBreakerMetricsProvider = new CircuitBreakerMetricsProviderImpl(actorSystem)
+      override lazy val serviceInfo: ServiceInfo = ServiceInfo("conductr-service.-est", Map.empty[String, immutable.Seq[ServiceAcl]])
+      override lazy val httpFilters = Seq.empty
+      override def withCircuitBreaker[T](id: String)(body: => Future[T]): Future[T] = body // @TODO this certainly can't be right?
```

Comment:
I have no idea what a proper implementation of this is for a test.

Reply:
I think it is reasonable to return `body` in the test since we are not testing for any circuit breaker related failure.

I'd expect the circuit breaker failure scenario for a given will be tested within Lagom itself.

If we were to test the failure scenario, I'd imagine during failure scenario `withCircuitBreaker` will return a failed `Future`.

Reply:
If Lagom team also thinks it's reasonable, we need to remove the `TODO`.

Reply:
I think the best approach is to:

1. Make `ConductRServiceLocatorComponents` declare an abstract `def circuitBreakersPanel: CircuitBreakersPanel` member
2. Make `app` in this test mix in `CircuitBreakerComponents` from Lagom
3. Delete `withCircuitBreaker` here